core/src/test/java/it/pagopa/selfcare/mscore/core/DelegationServiceImplTest.java

```diff
@@ -44,10 +45,36 @@ class DelegationServiceImplTest {
     void testCreateDelegation() {
         Delegation delegation = new Delegation();
         delegation.setId("id");
+        delegation.setProductId("prod-io");
+        Institution institution = new Institution();
+        institution.setId("id");
+        when(delegationConnector.save(any())).thenReturn(delegation);
+        doNothing().when(mailNotificationService).sendMailForDelegation(any(), any(), any());
+        doNothing().when(institutionService).updateInstitutionDelegation(any(),anyBoolean());
+        Delegation response = delegationServiceImpl.createDelegation(delegation);
+        verify(delegationConnector).save(any());
+        assertNotNull(response);
+        assertNotNull(response.getId());
+        assertEquals(delegation.getId(), response.getId());
+    }
+
+    /**
+     * Method under test: {@link DelegationServiceImpl#createDelegation(Delegation)}
+     */
+    @Test
+    void testCreateDelegationForPTWithProductPagopa() {
+        Delegation delegation = new Delegation();
```

Comment:
We could avoid to duplicate creation of delegation object in each method creating a static dummy object.

Reply:
done